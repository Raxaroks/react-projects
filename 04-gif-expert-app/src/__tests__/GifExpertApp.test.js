import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp />', () => {
    
    test('Debe de renderizarse correctamente.', () => {
        const componentWrapper = shallow(<GifExpertApp/>);
        expect(componentWrapper).toMatchSnapshot();
    })
    
    test('Debe de mostrar una lista de categorÃ­as.', () => {
        const ctgrs = ['Death Note', 'Naruto Shippuden', 'Excel Saga'];
        const componentWrapper = shallow(<GifExpertApp defaultCategories={ctgrs}/>);

        expect(componentWrapper).toMatchSnapshot();
        expect(componentWrapper.find('GifGrid').length).toBe(ctgrs.length);
    })
    
})
