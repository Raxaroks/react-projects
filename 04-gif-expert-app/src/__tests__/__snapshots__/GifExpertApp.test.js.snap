// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp /> Debe de mostrar una lista de categor√≠as. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Death Note"
      key="Death Note"
    />
    <GifGrid
      category="Naruto Shippuden"
      key="Naruto Shippuden"
    />
    <GifGrid
      category="Excel Saga"
      key="Excel Saga"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp /> Debe de renderizarse correctamente. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
